contributors/design-proposals/volume-provisioning-configuration.md

```diff
@@ -0,0 +1,207 @@
+    * PV controller would ignore existing PVs and it would **always provision a new PV** for a PVC with non-empty `spec.provisioningOptions` - we do not have any way how to check if an existing PV conforms to `spec.provisioningOptions` so dynamic provisioning is the only way.
+    * All provisioners (both internal and external) must provision a PV that satisfies both `StorageClass.parameters` and `PVC.spec.provisioningOptions` or throw an error if such combination of options is not possible.
+    * When user does not specify any `provisioningOptions` or the referred `ConfigMap` does not specify all allowed parameters, the provisioner either uses some defaults or throws an error that some parameters in the ConfigMap are mandatory.
+
+* StorageClass will get a new field to tell regular users what are allowed options in `PVC.spec.provisioningOptions`:
+    ```go
+    type StorageClass struct {
+        // <snip>
+
+        // AllowedProvisioningOptions contains details about provisioning options that are allowed by this StorageClass.
+        AllowedProvisioningOption  []ProvisioningOptionSpec
+    }
+
+    // ProvisioningOption describes one option that can be used in PVC provisioningOptions
+    // when requesting a volume of a particular StorageClass.
+    type ProvisioningOptionSpec struct {
+        // Name of key in ConfigMap
+        Key string
+
+        // Optional default value if it is unspecified in the ConfigMap.
+        DefaultValue *string
+
+        // Optional flag whether this selector key is mandatory in the ConfigMap.
+        // Defaults to "false".
+        Required bool
+
+        // Optional list of values that are allowed for this key in the ConfigMap.
+        AllowedValues []string
+
+        // Optional description. May contain newlines and links (e.g. URL of description of the option that would respect browser's Accept-Language).
+        Description *string
+    }
+    ```
+
+  * It's responsibility of an admin to fill `sc.allowedProvisioningOptions` when creating a StorageClass!
+  * We will update examples for all supported internal provisioners with selectorOptions for easier copy/paste.
+  * `kubectl describe storageclass` will show available options ad values in a user-friendly form.
+
+* There is no code in Kubernetes that validates user supplied `ConfigMap` with `allowedProvisioningOptions` in StorageClasses. Users may add invalid option keys or invalid values to their `ConfigMaps` and it's up to the individual provisioner to validate the `ConfigMap` and throw appropriate errors.
+
+## Examples
+
+### Ultimate StorageClass for AWS
+This example shows a rich StorageClass with three user-tunable parameters.
+
+```yaml
+apiVersion: storage.k8s.io/v1beta1
+kind: StorageClass
+metadata:
+  name: slow
+provisioner: kubernetes.io/aws-ebs
+parameters:
+  type: gp2
+  zones: us-east-1a, us-east-1b, us-east-1c, us-east-2a, us-east-2b, us-east-2c
```

Comment:
i see below you mean the admin restricts to these zones. if this doesnt repeated everything in the allowed, it will look more realistic example. My question about is there a default still stands ?